src/psyclone/tests/dynamo0p3_haloex_test.py

```diff
@@ -0,0 +1,405 @@
+    config = Config.get()
+    dyn_config = config.api_conf("dynamo0.3")
+    monkeypatch.setattr(dyn_config, "_compute_annexed_dofs", True)
+
+    # parse and get psy schedule
+    _, info = parse(os.path.join(BASE_PATH,
+                                 "14.13_halo_inc_to_inc.f90"),
+                    api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3").create(info)
+    schedule = psy.invokes.invoke_list[0].schedule
+
+    # check we have no halo exchanges for field "f1"
+    assert len(schedule.children) == 3
+    haloex = schedule.children[0]
+    loop1 = schedule.children[1]
+    loop2 = schedule.children[2]
+    assert isinstance(haloex, DynHaloExchange)
+    assert haloex.field.name == "f2"
+    assert haloex.required() == (True, False)
+    assert haloex._compute_halo_depth() == "1"
+    assert isinstance(loop1, DynLoop)
+    assert isinstance(loop2, DynLoop)
+
+    # just check compilation here (not later in this test) as
+    # compilation of redundant computation is checked separately
+    if utils.TEST_COMPILE:
+        # If compilation testing has been enabled
+        # (--compile --f90="<compiler_name>" flags to py.test)
+        assert utils.code_compiles("dynamo0.3", psy, tmpdir, f90, f90flags)
+
+    # make 1st loop iterate over cells to the level 2 halo and check output
+    rc_trans = Dynamo0p3RedundantComputationTrans()
+    rc_trans.apply(schedule.children[1], depth=2)
+
+    def check(schedule, f1depth, f2depth):
+        '''check that the schedule is modified in the expected way. In
+        particular, check that the depth of the halo exchange for
+        field 'f1' is what we are expecting
+
+        '''
+        assert len(schedule.children) == 4
+        haloex1 = schedule.children[0]
+        haloex2 = schedule.children[1]
+        loop1 = schedule.children[2]
+        loop2 = schedule.children[3]
+        assert isinstance(haloex1, DynHaloExchange)
+        assert haloex1.field.name == "f2"
+        assert haloex1._compute_halo_depth() == f2depth
+        assert haloex1.required() == (True, False)
+        assert isinstance(haloex2, DynHaloExchange)
+        assert haloex2.field.name == "f1"
+        assert haloex2._compute_halo_depth() == f1depth
+        assert haloex2.required() == (True, False)
+        assert isinstance(loop1, DynLoop)
+        assert isinstance(loop2, DynLoop)
+
+    # we should now have a speculative halo exchange at the start of
+    # the schedule for "f1" to depth 1 and "f2" to depth 2
+    check(schedule, f1depth="1", f2depth="2")
+
+    # make 1st loop iterate over cells to the maximum halo depth and
+    # check output
+    rc_trans.apply(schedule.children[2])
+    # we should now have a speculative halo exchange at the start of
+    # the schedule for "f1" to depth max halo - 1 and "f2" to max halo
+    check(schedule, f1depth="mesh%get_halo_depth()-1",
+          f2depth="mesh%get_halo_depth()")
+
+
+def test_gh_inc_nohex_4(tmpdir, f90, f90flags, monkeypatch):
```

Comment:
No I don't think it would be easy to merge as the number and behaviour of halo exchanges is different which leads to different transformations and tests.